app/controllers/editions_controller.rb

```diff
@@ -6,6 +6,9 @@ class EditionsController < InheritedResources::Base
   defaults resource_class: Edition, collection_name: "editions", instance_name: "resource"
   before_action :setup_view_paths, except: %i[index new create]
   before_action :require_editor_permissions, only: %i[update duplicate progress]
+  before_action only: %i[unpublish process_unpublish] do
+    require_govuk_editor(redirect_path: edition_path(resource))
```

Comment:
I see your point. It could certainly be done that way but when rebasing the commit that introduced this method we already knew we'd need the flexibility so we added it right from the start (just to avoid changing the method later on).

Reply:
Fair enough.